src/spikeinterface/preprocessing/deepinterpolation/generators.py

```diff
@@ -41,16 +34,16 @@ def __init__(
                 r.get_num_channels() == recordings[0].get_num_channels() for r in recordings[1:]
             ), "All recordings must have the same number of channels"
 
-        total_num_samples = np.sum(r.get_total_samples() for r in recordings)
+        total_num_samples = np.sum([r.get_total_samples() for r in recordings])
```

Comment:
Is there any chance this could overflow with numpy operations?  total sample number can get to be pretty big.

Same comment for the other instances.

Reply:
This is good to know though. I think the issue is data gets bigger and bigger so it's more and more likely to have overflows. I think we should be safe too. These are hard to debug when they go wrong.

Reply:
@h-mayorquin @zm711 so what do you recommend to do? I think you have a better overview of this problem than me :)